### PR TITLE
Include a closure for filtering searchable items (i.e. conditionally …

### PR DESCRIPTION
**Pull request by Harti**
…exclude them from being indexed)

See #153

### CODE REVIEW

**Comment by Harti** [comment]
Oh shoot, this is no default Builder object. My bad, nevermind.

File: src/Searchable.php [hunk 1]
```diff
@@ -99,6 +109,10 @@ public static function makeAllSearchable()
 
         $self->newQuery()
             ->orderBy($self->getKeyName())
+            ->filter(function($model)
```
Comment (Harti) [inline src/Searchable.php]: there should probably be a `->get()` beforehand

**Comment by Harti** [comment]
On another note, @taylorotwell, while I do appreciate all the work you do, documentation on how to properly test code changes before making a pull request is really poor on the contribution guide. 

I was unable to properly adapt Mockery to mock using `->get()` to return a collection which would then be `filter()`ed and lastly the `searchable()` method be applied to the results. Well, I _was_ capable of editing the test case. But I'm pretty certain that's not how it is supposed to go.

Some PHP "peasants", myself included, are probably hesitant to push changes as they don't feel professional enough to make properly tested and easily integratable pull requests. 

I do actually think that my proposed change would only have to be adapted a tad bit to work swiftly with Laravel, but contributions like mine just won't find their way into your software if rookie developers fail to run a successful CI test. Just my two cents, I don't care if this makes it into any Scout release or not, I can always override the trait like I did in #153, for me it's more about sharing the knowledge and making it easier for all the other folks to achieve the same thing.

**Comment by michelecurletta** [comment]
@taylorotwell what do you think about merging this PR?

**Pull request closed without merging**